client/chat/action-creators.ts

```diff
@@ -70,7 +70,7 @@ export function getMessageHistory(channel: string, limit: number): ThunkAction {
     }
 
     const chanData = byName.get(lowerCaseChannel)!
-    const earliestMessageTime = chanData.messages.first({ time: -1 }).time
+    const earliestMessageTime = chanData.messages[0]?.time ?? -1
```

Comment:
This code is too clever, prefer a ternary where you check the length and only access the first index if you're sure data is there